sources/util.go

```diff
@@ -244,3 +246,32 @@ func mustParseInt(value string) int {
 }
 
 var accept = "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8"
+
+func unzipInMemory(ctx context.Context, body []byte) ([]byte, error) {
+	log := app.Log.From(ctx)
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+	if err != nil {
+		log.Error().Msg(err.Error())
+		return nil, err
+	}
+
+	// Read all the files from zip archive
+	for _, zipFile := range zipReader.File {
+		unzippedFileBytes, err := readZipFile(zipFile)
+		if err != nil {
+			log.Error().Msg(err.Error())
+			continue
+		}
+		return unzippedFileBytes, nil
+	}
+	return nil, nil
```

Comment:
This is confusing: result is nil and error is nil. It must not happen. Return nil, fmt.errorf("zip: empty file") is better